src/org/omegat/core/search/Searcher.java

```diff
@@ -628,7 +630,24 @@ public boolean searchString(String origText, boolean collapseResults) {
             }
             while (true) {
                 int start = matcher.start();
-                foundMatches.add(new SearchMatch(start, matcher.end()));
+                if (m_searchExpression.mode == SearchMode.REPLACE && m_searchExpression.searchExpressionType == SearchExpression.SearchExpressionType.REGEXP) {
+                    String repl = m_searchExpression.replacement;
+                    Matcher replaceMatcher = Pattern.compile("(?<!\\\\)\\$(\\d+)").matcher(repl);
+                    while (replaceMatcher.find()) {
+                        int varId = Integer.parseInt(replaceMatcher.group(1));
+                        repl = repl.substring(0, replaceMatcher.start())
+                            + matcher.group(varId) // yes, from source matcher!
+                            + repl.substring(replaceMatcher.end());
+                        replaceMatcher.reset(repl);
+                    }
+                    if (!repl.equals(m_searchExpression.replacement)) {
+                        foundMatches.add(new ReplaceMatch(start, matcher.end(), replaceCase(repl)));
```

Comment:
`replaceCase` is only used here, when the replacement string is match-specific (i.e. when there was a replacement). I suppose it doesn't make much sense for a user to want use case transformations on a constant replacement string, but it feels inconsistent to me that the `\u`, etc., operators would behave as literals when there's no backreference.

How is this handled in other tools? If it's standard behavior to become literals in the absence of backreferences then I guess we should do so as well.

Reply:
In Perl, or in Bash, there is the clear distinction between interpolated strings (in double quotes) and non-interpolated ones (in simple quotes). When a string is interpolated, case operators _are_ applied even if there was finally no variable in it (note that I do not speak about regular expressions here, only about interpolation of variables, $1 or $x for example)
Sorry to make reference to our fork, but in DGT-OmegaT specific replace screen, this is one of the reasons why it makes sense to have a checkbox to activate or not the interpolation: when inactivated, $1 remains $1 and \L remains \L, as is. Only users which are really familiar with regex replacements should activate it.

Reply:
OK sorry I understood later why you wrote that. Yes, you are true, \L should always be interpreted when we are in regex mode.
With the new code the rule is: always interpreted in regex mode, never interpreted in "exact search" mode.
However you should consider the opportunity to add a checkbox for optional support of regex variables: only users who really know what they are doing should use it.